### DISCUSSION TITLE
从2更新到3后，回调函数里，提示response 的类型为unkown，之前没有

### DISCUSSION THREAD

**Discussion by manortec** [alovajs/alova#478] [Q&A] [answered]
login函数是我包装的一个函数如下：
![QQ20240724-130405](https://github.com/user-attachments/assets/3b49f2bd-e20f-4883-ae2b-bbfab7c39f3b)
从2更新到3后，下面函数标红提示类型未知错误，但是实际运行是正确的。
![QQ20240724-130503](https://github.com/user-attachments/assets/b84ef002-8e37-4979-bbda-712081a9c4ea)
拦截器里返回的response是 
![图片](https://github.com/user-attachments/assets/086f5d2b-8032-4ee2-acdd-35f1bb281d4f)

这可能并不算一个问题，只是更新后vscode里一直标红，很别扭，看看是否有优化的空间，谢谢！

**Comment by JOU-amjs** [comment, marked as answer]
在ts中使用的话，`alovaInstance.Post`没有指定返回数据类型默认就是`unknown`，这其实也是在提示你可能忘记指定数据类型了，如果你不想指定则给它一个any即可，像这样`alovaInstance.Post<any>(...)`

**Reply by graphsay** [reply]
向你请教：我不是用Typescript，我用的是jsdoc来定义类型，我查了不少资料，实在不太清楚怎么样对Post的泛型参数进行类型注释，或者说在其他位置可以进行类型注释，以消除vscode的这个错误提示，或者说将其默认设定为any类型也可以，感谢给些意见，谢谢

**Reply by MeetinaXD** [reply]
我建议你手动重新导出每一个alova api函数，并编写单独的d.ts文件声明其类型。这可能是\**不使用\** typescript 的情况下最好的方法。

**Reply by graphsay** [reply]
谢谢建议，我尝试一下
